designer/client/src/components/modals/ValidationLabels.tsx

```diff
@@ -54,3 +68,64 @@ export default function ValidationLabels(props: Props) {
         </LabelsContainer>
     );
 }
+
+type TemplateLabelProps = {
+    errorCode: string;
+    templateValues: TemplateValues;
+};
+
+const TemplatedType = styled.span({
+    fontWeight: "bold",
+    textDecoration: "underline",
+    "&:hover": {
+        color: "white",
+    },
+});
+
+function I18nTemplateLabel(props: TemplateLabelProps) {
+    const { errorCode, templateValues } = props;
+    const { i18n } = useTranslation();
+
+    const displayableTemplateValues = useMemo(() => {
+        return {
+            ...templateValues.stringValues,
+            ...Object.fromEntries(
+                Object.entries(templateValues.typingResultValues).map(([key, typingResult]) => {
+                    return [key, typingResult.display];
+                }),
+            ),
+        };
+    }, [templateValues]);
+
+    // we create separate spans only for TypingResult values
+    const templateComponents = useMemo(() => {
+        return Object.fromEntries(
+            Object.entries(templateValues.typingResultValues).map(([key, typingResult]) => {
+                const component = <TemplatedType title={typingResult.type} />;
+                return [key, component];
+            }),
+        );
+    }, [templateValues]);
+
+    const canRender =
+        i18n.exists(`nodeErrorCodes:${errorCode}.messageTemplate`) &&
+        validateTemplateInterpolation(
+            i18n.t(`nodeErrorCodes:${errorCode}.messageTemplate`),
+            Object.keys(displayableTemplateValues),
+            Object.keys(templateComponents),
+        );
+
+    return canRender ? (
```

Comment:
IMO, This check should only warn us somehow (for example console warn). We should also have e2e tests to check if we are able to render specific template variants (not sure how many variants we will have). Thanks to it we will be able to verify if someone not broke the contract.

Reply:
Agree, with proper testing and contract this will not be needed.